XCTestBootstrap/TestManager/FBTestManagerTestReporterJUnit.h

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "FBTestManagerTestReporter.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ A Test Reporter that implements the FBTestManagerTestReporter interface.
+ It writes the Test Result to a given File Handle in the JUnit XML format.
+ */
+@interface FBTestManagerTestReporterJUnit : NSObject <FBTestManagerTestReporter>
+
+/**
+ Constructs a JUnit Test Reporter.
+
+ @param outputFileHandle a file handle the JUnit XML should be written to.
+ */
++ (instancetype)withOutputFileHandle:(NSFileHandle *)outputFileHandle;
```

Comment:
Is the main reason for using a file handle here that you can stream the results to a pipe instead of a file?

It does add a burden to the consumer of the API to ensure that the file handle is closed. If you only care about files you could just give a file path and manage the file handle internally to the class. The reporter could also clean the file handle on `dealloc` or when there is a final event in the `FBTestManagerTestReporter` protocol indicating that the stream should be closed.